scripts/generate-framework-agreement-counterpart-signature-pages.py

```diff
@@ -95,17 +98,29 @@
     html_dir = tempfile.mkdtemp()
 
     records = find_suppliers_with_details_and_draft_service_counts(client, framework_slug, supplier_ids)
+    records = list(records)
+    for record in records:
+        if framework_supports_e_signature(record['frameworkSlug']) and \
+                (client.get_framework_agreement(record['agreementId'])
+                 ['agreement']['status'] == 'signed'):
+            # E-signatures are automatically approved.
+            client.approve_agreement_for_countersignature(record['agreementId'],
```

Comment:
Hmm, to be honest I am still not 100% sure of the logic either, but I _think_ and judging by the previous Jenkins runs. We may be protected by this line https://github.com/alphagov/digitalmarketplace-scripts/blob/27288765a66bcb50c568658c6d14a91489546fbe/dmscripts/generate_framework_agreement_signature_pages.py#L104-L105
i.e. once the `countersigned_path` is set (which should be done by the upload script) it skips over the supplier. 
We should definitely test this on preview though.

Reply:
@gidsg is correct, once a PDF has been uploaded the generate script will not create a new one

Reply:
that means that if the supplier signs again, we won't countersign again (which I think was something we said it would do?). In any case, I think we can off-scope that, as it's new behaviour that wasn't there before.

Reply:
@phantas I _believe_ that when a countersign approval happens it changed the status from `signed` to `approved`. If they re-sign it reverts the status to `signed` which would allow the re-approval to take place.

Reply:
If they re-sign it will generate a new agreement, which will not have `countersigned_path` set.

Reply:
ah great, thanks!